app/src/main/java/com/practicum/playlistmaker/SettingsActivity.kt

```diff
@@ -14,15 +19,51 @@ class SettingsActivity : AppCompatActivity() {
         setContentView(R.layout.activity_settings)
 
         findViewById<TextView>(R.id.back).setOnClickListener {
-            val intent = Intent(this, MainActivity::class.java)
-            //startActivity(intent) Для выхода с экрана стоит использовать функцию finish(),
-            // а не startActivity, что создаст новую активити и откроет её.
-            // Принципиальная разница заключается в том, что в твоём коде вместо
-            // возврата открывается и создаётся новый экран с новым состоянием,
-            // а при использовании finish() закрывается текущий экран и показывается
-            // предыдущий, который был открыт перед текущим.
             finish()
         }
 
+        val btnShare = findViewById<TextView>(R.id.btnShare)
+        btnShare.setOnClickListener {
+            shareApp()
+        }
+
+        val supportButton = findViewById<TextView>(R.id.supportButton)
+        supportButton.setOnClickListener {
+            sendEmail()
+        }
+        val userAgreementButton = findViewById<TextView>(R.id.userAgreementButton)
+        userAgreementButton.setOnClickListener {
+            openUserAgreement()
+        }
+    }
+    private fun shareApp() {
+        val shareText = getString(R.string.share_text)
+        val intent = Intent(Intent.ACTION_SEND)
+        intent.type = "text/plain"
+        intent.putExtra(Intent.EXTRA_TEXT, shareText)
+        startActivity(Intent.createChooser(intent, getString(R.string.choose_app)))
+    }
+
+    private fun sendEmail() {
+        val emailIntent = Intent(Intent.ACTION_SENDTO)
+        emailIntent.data = Uri.parse("mailto:" + getString(R.string.support_email))
```

Comment:
Тут не факт, что этот метод сработает, например с Gmail он точно не работает. Нужно так, так как ожидается не просто строка, а массив строк-получателей:
```suggestion
        emailIntent.data = "mailto:".toUri()
        emailIntent.putExtra(Intent.EXTRA_EMAIL, arrayOf(getString(R.string.support_email)))
```